scylla/src/routing.rs

```diff
@@ -6,7 +6,23 @@ use thiserror::Error;
 
 #[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Debug)]
 pub struct Token {
-    pub value: i64,
+    value: i64,
+}
+
+impl Token {
+    pub fn new(value: i64) -> Self {
+        Self {
+            value: if value == i64::MIN { i64::MAX } else { value },
+        }
+    }
+
+    pub fn new_denormalized(value: i64) -> Self {
+        Self { value }
+    }
+
+    pub fn value(&self) -> i64 {
+        self.value
+    }
 }
```

Comment:
What about another approach?
`value` stores the original value passed to `new`, then we have two getters:
- `value()`, which performs normalisation wrt `i64::MIN`,
- `value_denormalized()`, which returns `value` without normalisation.

With this approach, we could make `value_denormalized()` a `pub(crate)` method.

Reply:
The only drawback I can see is a minor (negligible) overhead upon each access to the value.

Reply:
Normalization / denormalization seems like something that should be decided upon creation, not upon access.
Look at the code changed: in cdc partitioner we create denormalized token. Let's say I apply your suggestion. Which uses of `value()` should be changed to `value_denormalized()`? All of them?

Reply:
You're right, I haven't noticed the exact use in `CDCPartitioner`, which fully justifies the proposed semantics.